src/shared/components/dataAccessTokens/DataAccessTokens.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from "react";
+import AppConfig from "appConfig";
+import {If, Then, Else} from 'react-if';
+import internalClient from "shared/api/cbioportalInternalClientInstance";
+import {AppStore} from "../../../AppStore";
+import LoadingIndicator from "../loadingIndicator/LoadingIndicator";
+import {observer} from "mobx-react";
+import {observable} from "mobx";
+import {buildCBioPortalPageUrl} from "../../api/urls";
+import {isNullOrUndefined} from "util";
+import fileDownload from 'react-file-download';
+
+export class UserDataAccessToken {
+    @observable token : string;
+    @observable creationDate: string;
+    @observable expirationDate: string;
+    @observable username: string;
+    constructor(token: string, creationDate: string, expirationDate: string, username: string) {
+        this.token = token;
+        this.creationDate = creationDate;
+        this.expirationDate = expirationDate;
+        this.username = username;
+    }
+}
+
+export interface IDataAccessTokensProps {
+    token?:string;
+    creationDate?:string;
+    expirationDate?:string;
+    loadingComponent?:JSX.Element;
+    appStore:AppStore;
+}
+
+function buildDataAccessTokenFileContents(dat:UserDataAccessToken | undefined) {
+    if (!isNullOrUndefined(dat)) {
+        const fileContents = "token: " + dat!.token + "\n" +
+            "creation_date: " + new Date(dat!.creationDate).toISOString()+ "\n" +
+            "expiration_date: " + new Date(dat!.expirationDate).toISOString();
+        return fileContents;
+    } else {
+        alert("Cannot create Data Access Token file for user with non-existent tokens.");
+        return null;
+    }
+}
+
+@observer
+export class DataAccessTokens extends React.Component<IDataAccessTokensProps, {}> {
+    public static defaultProps: Partial<IDataAccessTokensProps> = {
+        loadingComponent:<LoadingIndicator isLoading={true}/>
+    };
+
+    constructor(props: IDataAccessTokensProps) {
+        super(props);
+    }
+
+    getDatDropdownList() {
+        const listItems = [
+            {
+                id:"signout",
+                action:<a href={buildCBioPortalPageUrl(this.props.appStore.logoutUrl)}>Sign out</a>,
+                hide:false
+            },
+            {
+                id:"datDownload",
+                action:<a onClick={() => this.downloadDataAccessTokenFile()}>Download token</a>,
+                hide:(AppConfig.serverConfig.authenticationMethod === "social_auth" || (AppConfig.serverConfig.dat_method !== "uuid" && AppConfig.serverConfig.dat_method !== "jwt"))
+            }
+        ];
+        const shownListItems = listItems.filter((l)=>{
```

Comment:
signout item will never be shown!?

Reply:
it shows @alisman :)  `hide = false` for `signout`

Reply:
@alisman the `datDownload` item will only get filtered out if authentication method is not `social_auth` or if the DAT configuration is not `uuid` or `jwt`